plugins/inputs/webhooks/github/github_webhooks.go

```diff
@@ -105,3 +116,14 @@ func NewEvent(data []byte, name string) (Event, error) {
 	}
 	return nil, &newEventError{"Not a recognized event type"}
 }
+
+func checkSignature(secret string, data []byte, signature string) bool {
+	return hmac.Equal([]byte(signature), []byte(generateSignature(secret, data)))
+}
+
+func generateSignature(secret string, data []byte) string {
```

Comment:
The generation is dependant of the body of the event. Not sure what I can cache. The string to byte op?